src/lib/support/PersistentData.h

```diff
@@ -49,13 +49,23 @@ struct PersistentData
         TLV::TLVWriter writer;
         writer.Init(mBuffer, sizeof(mBuffer));
 
-        ReturnErrorOnFailure(Serialize(writer));
-
-        // Save serialized data
-        return storage->SyncSetKeyValue(key.KeyName(), mBuffer, static_cast<uint16_t>(writer.GetLengthWritten()));
+        CHIP_ERROR err = Serialize(writer);
+        if (CHIP_NO_ERROR == err)
+        {
+            // Save serialized data
+            err = storage->SyncSetKeyValue(key.KeyName(), mBuffer, static_cast<uint16_t>(writer.GetLengthWritten()));
+        }
+        return err;
     }
 
-    virtual CHIP_ERROR Load() { return this->Load(this->mStorage); }
+    virtual CHIP_ERROR Load(bool force = false)
```

Comment:
So the fact that Load() by default does not force means ops on PersistentArray with an entry type that does not own is own data (the only kind we have so far) will lead to a mix of entries pointing into our TLV buffer and entries pointing to caller-provided buffers, no?

Reply:
Yes. In the case of PersistentArray, after a load, all pointers point to the internal buffer. Those pointer may be modified to point to caller-provider buffers. However, this behavior holds regardless the force flag, which is used to prevent unnecessary NVM reads.
I have changed the default to true, which I think is expected when calling this method directly without argument.